cluster/gce/gci/scaleoutserver-helper.sh

```diff
@@ -110,3 +110,90 @@ function create-proxy-instance-internal() {
+    # Workaround for #55777
+    retries=30
+    sleep_sec=60
+  fi
+
+  local -r server_name="${1}"
+  local -r address="${2:-}"
+  local -r private_network_ip="${3:-}"
+
+  local enable_ip_aliases
+  if [[ "${NODE_IPAM_MODE:-}" == "CloudAllocator" ]]; then
+    enable_ip_aliases=true
+  else
+    enable_ip_aliases=false
+  fi
+
+  local network=$(make-gcloud-network-argument \
+    "${NETWORK_PROJECT}" "${REGION}" "${NETWORK}" "${SUBNETWORK:-}" \
+    "${address:-}" "${private_network_ip:-}" "${enable_ip_aliases:-}" "${IP_ALIAS_SIZE:-}")
+
+  local metadata="kube-env=${KUBE_TEMP}/master-kube-env.yaml"
+  metadata="${metadata},kubelet-config=${KUBE_TEMP}/master-kubelet-config.yaml"
+  metadata="${metadata},user-data=${KUBE_ROOT}/cluster/gce/gci/master.yaml"
+  metadata="${metadata},configure-sh=${KUBE_ROOT}/cluster/gce/gci/configure.sh"
+  metadata="${metadata},apiserver-config=${KUBE_ROOT}/hack/apiserver.config"
+  metadata="${metadata},cluster-location=${KUBE_TEMP}/cluster-location.txt"
+  metadata="${metadata},cluster-name=${KUBE_TEMP}/cluster-name.txt"
+  metadata="${metadata},gci-update-strategy=${KUBE_TEMP}/gci-update.txt"
+  metadata="${metadata},gci-ensure-gke-docker=${KUBE_TEMP}/gci-ensure-gke-docker.txt"
+  metadata="${metadata},gci-docker-version=${KUBE_TEMP}/gci-docker-version.txt"
+  metadata="${metadata},kube-master-certs=${KUBE_TEMP}/kube-master-certs.yaml"
+  metadata="${metadata},cluster-location=${KUBE_TEMP}/cluster-location.txt"
+  metadata="${metadata},controllerconfig=${KUBE_TEMP}/controllerconfig.json"
+  metadata="${metadata},${MASTER_EXTRA_METADATA}"
+
+  local disk="name=${server_name}-pd"
+  disk="${disk},device-name=master-pd"
+  disk="${disk},mode=rw"
+  disk="${disk},boot=no"
+  disk="${disk},auto-delete=no"
+
+  for attempt in $(seq 1 ${retries}); do
+    if result=$(${gcloud} compute instances create "${server_name}" \
+      --project "${PROJECT}" \
+      --zone "${ZONE}" \
+      --machine-type "${SCALEOUTSERVER_SIZE:-${MASTER_SIZE}}" \
+      --image-project="${SCALEOUTSERVER_IMAGE_PROJECT:-${MASTER_IMAGE_PROJECT}}" \
+      --image "${SCALEOUTSERVER_IMAGE:-${MASTER_IMAGE}}" \
+      --tags "${server_name}" \
+      --scopes "storage-ro,compute-rw,monitoring,logging-write" \
+      --metadata-from-file "${metadata}" \
+      --disk "${disk}" \
+      --boot-disk-size "${SCALEOUTSERVER_ROOT_DISK_SIZE:-${MASTER_ROOT_DISK_SIZE}}" \
+      ${SCALEOUTSERVER_MIN_CPU_ARCHITECTURE:+"--min-cpu-platform=${SCALEOUTSERVER_MIN_CPU_ARCHITECTURE}"} \
+      ${network} 2>&1); then
+      echo "${result}" >&2
+      return 0
+    else
+      echo "${result}" >&2
+      if [[ ! "${result}" =~ "try again later" ]]; then
+        echo "Failed to create scaleoutserver instance due to non-retryable error" >&2
+        return 1
+      fi
+      sleep $sleep_sec
+    fi
+  done
+
+  echo "Failed to create scaleoutserver instance despite ${retries} attempts" >&2
```

Comment:
please log out the instance name etc for debugging purpose.

Reply:
will add in follow-up optimized PR